Tasks/Common/azure-blobstorage-downloader/azureBlobStorageProvider.ts

```diff
@@ -0,0 +1,139 @@
+            resolve(items);
+        });
+    }
+
+    public getArtifactItem(artifactItem: models.ArtifactItem): Promise<Stream.Readable> {
+        return new Promise(async (resolve, reject) => {
+            var readStream: Stream.Readable = this._blobSvc.createReadStream(this._container, artifactItem.path, (error, result) => {
+                if (error) {
+                    console.log("Unable to fetch item: " + artifactItem.path + ". Error: " + error);
+                    reject();
+                }
+            });
+            resolve(readStream);
+        });
+    }
+
+    private _ensureContainerExistence(): Promise<void> {
+        return new Promise<void>(async (resolve, reject) => {
+            if (!this._isContainerExists) {
+                var self = this;
+                this._blobSvc.createContainerIfNotExists(this._container, function (error, result, response) {
+                    if (!!error) {
+                        console.log("Failed to create container " + self._container + ". Error: " + error.message);
+                        reject(error);
+                    } else {
+                        self._isContainerExists = true;
+                        console.log("Created container " + self._container);
+                        resolve();
+                    }
+                });
+            } else {
+                resolve();
+            }
+        });
+    }
+
+    private _getItems(container: string, parentRelativePath?: string): Promise<models.ArtifactItem[]> {
+        var promise = new Promise<models.ArtifactItem[]>(async (resolve, reject) => {
+            var items: models.ArtifactItem[] = [];
+            await this._ensureContainerExistence();
+
+            var self = this;
+            this._blobSvc.listBlobsSegmented(container, null, (error, result, response) => {
+                if (!!error) {
+                    console.log("Failed to list items inside container: " + self._container + ". Error: " + error.message);
+                    reject(error);
+                } else {
+                    console.log("Successfully fetcted list of items.");
+                    items = self._convertBlobResultToArtifactItem(result.entries);
+                    resolve(items);
+                }
+            });
+        });
+
+        return promise;
+    }
+
+    private _convertBlobResultToArtifactItem(blobResult: azureStorage.BlobService.BlobResult[]): models.ArtifactItem[] {
+        var artifactItems: models.ArtifactItem[] = new Array<models.ArtifactItem>();
+        blobResult.forEach(element => {
+            var artifactitem: models.ArtifactItem = new models.ArtifactItem();
+            artifactitem.itemType = models.ItemType.File;
+            artifactitem.fileLength = parseInt(element.contentLength);
+            artifactitem.lastModified = new Date(element.lastModified + 'Z');
+            artifactitem.metadata = element.metadata;
+            artifactitem.path = element.name;
```

Comment:
element.name includes sub-folders as well?

Reply:
this contains the path from container root till the blob locations including blobs name.